src/lib/services/addon-service.ts

```diff
@@ -186,10 +191,14 @@ export default class AddonService {
         return Promise.resolve();
     }
 
-    async createAddon(data: IAddonDto, userName: string): Promise<IAddon> {
+    async createAddon(data: ActualAddonDto, userName: string): Promise<IAddon> {
         const addonConfig = await addonSchema.validateAsync(data);
-        await this.validateKnownProvider(addonConfig);
-        await this.validateRequiredParameters(addonConfig);
+        try {
+            await this.validateKnownProvider(addonConfig);
+            await this.validateRequiredParameters(addonConfig);
+        } catch (e) {
+            throw new BadDataError(e);
+        }
```

Comment:
This gets around some API errors where the response is either 500 or not clear (I forget which one)